recommendation/src/main/java/com/yas/recommendation/topology/ProductTopology.java

```diff
@@ -0,0 +1,359 @@
+    private static final String PRODUCT_ATTRIBUTE_AGGREGATION_MATERIALIZED_VIEW = "prod-attr-agg-mv";
+    private static final String PRODUCT_WITH_ALL_DETAIL_MATERIALIZED_VIEW = "product-with-all-detail-mv";
+
+    @Autowired
+    public ProductTopology(KafkaTopicConfig kafkaTopicConfig) {
+        super(kafkaTopicConfig);
+    }
+
+    @Override
+    @Autowired
+    protected void process(StreamsBuilder streamsBuilder) {
+        KTable<Long, ProductDto> productTable = createProductTable(streamsBuilder);
+        KTable<Long, BrandDto> brandTable = createBrandTable(streamsBuilder);
+        KTable<Long, ProductResultDto> productBrandTable = joinProductAndBrand(productTable, brandTable);
+
+        KTable<Long, CategoryDto> categoryTable = createCategoryTable(streamsBuilder);
+
+        KTable<Long, ProductCategoryDto> productCategoryTable =
+                createProductCategoryTable(streamsBuilder);
+
+        KTable<Long, ProductCategoryDto> nonNullProductCategoryTable =
+                filterNonNullProductCategory(productCategoryTable);
+
+        KTable<Long, ProductCategoryDto> productCategoryDetailTable =
+                enrichProductCategoryWithDetails(productCategoryTable, categoryTable);
+
+        KTable<Long, AggregationDto<CategoryDto>> productCategoryAgg =
+                aggregateProductCategoryDetails(productCategoryDetailTable, nonNullProductCategoryTable);
+
+        KTable<Long, ProductResultDto> addingCategoryDetailTable =
+                joinProductWithCategoryDetails(productBrandTable, productCategoryAgg);
+
+        KTable<Long, ProductAttributeDto> productAttributeTable = createProductAttributeTable(streamsBuilder);
+
+        KTable<Long, ProductAttributeValueDto> productAttributeValueTable =
+                createProductAttributeValueTable(streamsBuilder);
+
+        KTable<Long, ProductAttributeValueDto> nonNullProductAttributeValueTable =
+                filterNonNullProductAttributeValue(productAttributeValueTable);
+
+        KTable<Long, ProductAttributeValueDto> productAttributeValueDetailTable =
+                enrichProductAttributeValueDetails(productAttributeValueTable, productAttributeTable);
+
+        KTable<Long, AggregationDto<ProductAttributeDto>> productAttributeAgg =
+                aggregateProductAttributeDetails(productAttributeValueDetailTable, nonNullProductAttributeValueTable);
+
+        KTable<Long, ProductResultDto> resultTable =
+                joinProductWithAttributeDetails(addingCategoryDetailTable, productAttributeAgg);
+
+        writeToSink(resultTable);
+    }
+
+    private KTable<Long, ProductDto> createProductTable(StreamsBuilder streamsBuilder) {
+        return streamsBuilder.stream(
+                        kafkaTopicConfig.product(),
+                        Consumed.with(getSerde(KeyDto.class), getMessageDtoSerde(ProductDto.class)))
+                .selectKey((key, value) -> key.getId())
+                .mapValues(this::extractModelFromMessage)
```

Comment:
Do we need to handle hard delete record in case? If not I think we can filter them here.
```java
.filter((key, value) -> value == null || value.getAfter() != null)
```

Reply:
Ignore, this case we will store hard delete records.